### PR TITLE
Feature/location widget

### PR DESCRIPTION
**Pull request by adamjohnwright**
This pull request contains the location widget and bug fixes for the sequence widget.

I have been working with Chriss to make sure the data is being displayed correctly.

### CODE REVIEW

File: src/rest_api/classes/gene/widgets/location.clj [hunk 1]
```diff
@@ -0,0 +1,50 @@
+(ns rest-api.classes.gene.widgets.location
+  (:require
+    [rest-api.classes.gene.sequence :as sequence-fns]
+    [rest-api.classes.gene.generic :as generic]
+))
+
+(defn genetic-position [gene]
+  (let [segment (sequence-fns/get-longest-segment gene)
+        gene-map (:gene/map gene)
+        chr (if (nil? gene-map) nil (:map/id (:gene.map/map gene-map)))
```
Comment (sibyl229) [inline src/rest_api/classes/gene/widgets/location.clj]: @a8wright In clojure, using keyword on nil returns nil, as desired, so you don't need a special case here.  `(:map/id (:gene.map/map gene-map)` should be sufficient. So are the case with the next few lines.

**Review by mgrbyte — COMMENTED** [review]
Seems to be double-white-spacing going on in gene/sequence.clj.

File: src/rest_api/classes/gene/widgets/location.clj [hunk 2]
```diff
@@ -0,0 +1,50 @@
+(ns rest-api.classes.gene.widgets.location
+  (:require
+    [rest-api.classes.gene.sequence :as sequence-fns]
+    [rest-api.classes.gene.generic :as generic]
+))
```
Comment (mgrbyte) [inline src/rest_api/classes/gene/widgets/location.clj]: wrap parens up to prev line

File: src/rest_api/classes/gene/widgets/location.clj [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+           ["TRANSPOSONS"
+            "TRANSPOSON_GENES"]
+           ["GENES"
+            "VARIATIONS_CLASSICAL_ALLELES"
+            "CLONES"])
+   :description "tracks displayed in GBrowse"})
+
+(defn- genomic-obj [gene]
+ (if-let [segment (sequence-fns/get-longest-segment gene)]
+           (let [[start, stop] (->> segment
```
Comment (mgrbyte) [inline src/rest_api/classes/gene/widgets/location.clj]: indentation gone wonky here (inner `let [[start stop]]`) , plus comma not needed (although if helps other's read-ability I'm not against the comma)

File: src/rest_api/classes/gene/sequence.clj [hunk 4]
```diff
@@ -19,3 +13,60 @@
...
+                                   (double)
+                                   (* mult-offset)
+                                   (int)
+                                   (x-op x))
+                              y))
+        browser-start  (calc-browser-pos - start stop 0.2)
+        browser-stop  (calc-browser-pos + stop start 0.5)
+        label (str (:seqname segment) ":" start ".." stop)
+        id (str  (:seqname segment) ":" browser-start ".." browser-stop)]
+  (pace-utils/vmap
```
Comment (mgrbyte) [inline src/rest_api/classes/gene/sequence.clj]: think this vmap bit needs indenting-in by one level

**Review by mgrbyte — COMMENTED** [review]
just the one

**Review by mgrbyte — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**